### ISSUE TITLE
leave file when on pause?

### ISSUE THREAD

**Issue by ClaireCJS** [Aldaviva/WinampNowPlayingToFile#19]
Hi!  I really appreciated how incredibly responsive you were the other day, and how someone else already had a pull request for it.   I really had no idea that was a winamp plugin "scene" on github that I was unaware of. 

Anyway... Would it be possible to add a configuration checkbox (or even INI/.rc file or whatever) to change this one behavior:

When you pause or stop, the plugin erases now_playing.jpg

And I totally get that. The file is called now playing, and now, it is not playing, so it's technically incorrect. Makes totally sense why that decision was made.

However, I'd like to have the option to change that behavior, and leave the image in place 
when paused and stopped.   
[Maybe still delete them on winamp.close()??]

basically.. I'm using that file for my dashboard, and I don't want to have to cache the image because i want to keep the resource profile as low as possible, which is another reason I like using your plugin to get the album art. I'm positive it does it more efficiently than me doing it in non-compiled non-winamp-plugin python code :) Thus lightening the load for my winamp_dashboard project.

Might have to rename it though. It now detects vlcplayer too, if I've stopped winamp, and tells me that instead. So whether i'm listening to music or watching movies, the dashboard remains accurate.

**Comment by Aldaviva** [comment]
So you're interested in the image file containing the album art of the song that was last playing after you stop/pause? Right now the plugin can display a static "stopped" placeholder, but not the last playing artwork. This could be added as a new configurable feature.

**Comment by ClaireCJS** [comment]
> So you're interested in the image file containing the album art of the song that was last playing after you stop/pause? 

Very much so. It would be nice if there were options for:

* keep image while paused 
* keep image after stop 
* keep image after closing winamp 

I actually had to catch exceptions because every once in awhile the timing was such that the file would be there when i initialize an object, but not by the next line when I draw the object, because winamp stopped in that exact small time interval between.  That was how I realized the file wasn't always there in the first place. For some reason, I was mentally thinkning of it more as a "now/last playing" than just a "now playing". My bad, that's on me.

I actually published my [WinAmp Dashboard pproject](https://github.com/ClaireCJS/turing_smart_screen_winamp_dashboard) just now :)

**Comment by Aldaviva** [comment]
Makes sense, I will try to incorporate this into the next release.

Yeah there are definitely some race conditions you get exposed to when songs are changing. There was a good one when the playback state changed before the song data changed, so it looked like the previous song started playing again, which was a problem if the previous song had been deleted.

Nice, having the pictures in the readme definitely helps illustrate how your project works.

**Comment by ClaireCJS** [comment]
Hi! I just did some updates to my "code" at home and ran into this wall again.

Long story short, I previously gathered the information on which track was playing by autoparsing my last.fm logs. I just today switched to instead autoparsing the nowplaying.txt

Well, i paused the song cause it was BabyMetal and i couldn't think, lo and behold a script failed, i figured out the reason was this file doesn't exist because i pausedi t.  I came here to request a feature to leave it, and see that... I've been here before, already requested it, you already did it, and just released it 6 weeks ago.  Yayyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy thank youuuuuuuuuuuuuuuu!!!!! This is what I love about Github! Gonna try it out now!

**Comment by ClaireCJS** [comment]
Ohhhh that's just the commit, it's not a release.

Need a beta tester? I could really use that feature right now....

**Comment by Aldaviva** [comment]
Here is a build of the most recent commit. I've only tested it a little, so I would expect it to have problems.

[WinampNowPlayingToFile-3.0.0.zip](https://github.com/user-attachments/files/17837792/WinampNowPlayingToFile-3.0.0.zip)

**Comment by ClaireCJS** [comment]
> Here is a build of the most recent commit. I've only tested it a little, so I would expect it to have problems.
> 
> [WinampNowPlayingToFile-3.0.0.zip](https://github.com/user-attachments/files/17837821/WinampNowPlayingToFile-3.0.0.zip)

Thank you, I appreciate it! You'll have an early tester :)

**Comment by ClaireCJS** [comment]
> Here is a build of the most recent commit. I've only tested it a little, so I would expect it to have problems.

Alas, it's completely non-functional for me.  Every song change and upon opening, it throws this error:


![image](https://github.com/user-attachments/assets/43c393ba-0415-4c53-8e90-a703c97d528c)


I had to revert right away to be able to continue to use winamp without collecting error dialogs


Also, when i went into the config and changed the new dropdown that [looks like it] lets us have multiple files, i got:

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at System.Collections.Generic.List`1.set_Item(Int32 index, T value)
   at WinampNowPlayingToFile.Presentation.SettingsDialog.saveWorking() in C:\Users\Ben\Documents\Projects\WinampNowPlayingToFile\WinampNowPlayingToFile\Presentation\SettingsDialog.cs:line 264
   at WinampNowPlayingToFile.Presentation.SettingsDialog.onTextFileMenuSelectionChanged(Object sender, EventArgs e) in C:\Users\Ben\Documents\Projects\WinampNowPlayingToFile\WinampNowPlayingToFile\Presentation\SettingsDialog.cs:line 116
   at System.Windows.Forms.ComboBox.OnSelectedIndexChanged(EventArgs e)
   at System.Windows.Forms.ComboBox.WmReflectCommand(Message& m)
   at System.Windows.Forms.ComboBox.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9282.0 built by: NET481REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
gen_WinampNowPlayingToFile
    Assembly Version: 1.0.8703.38477
    Win32 Version: 
    CodeBase: file:///C:/util2/WinAmp%20repo/WinAmp/Plugins/gen_WinampNowPlayingToFile.dll
----------------------------------------
Daniel15.Sharpamp
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/util2/WinAmp%20repo/WinAmp/Daniel15.Sharpamp.DLL
----------------------------------------
WinampNowPlayingToFile
    Assembly Version: 3.0.0.0
    Win32 Version: 3.0.0.0
    CodeBase: file:///C:/util2/WinAmp%20repo/WinAmp/WinampNowPlayingToFile.DLL
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9251.0 built by: NET481REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9282.0 built by: NET481REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9037.0 built by: NET481REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
mustache-sharp
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/util2/WinAmp%20repo/WinAmp/mustache-sharp.DLL
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9277.0 built by: NET481REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9037.0 built by: NET481REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9037.0 built by: NET481REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Accessibility
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9037.0 built by: NET481REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.


```

Guess i'll have to be more patient for the new release!

**Comment by Aldaviva** [comment]
I'm looking at this again. You're right, it was incredibly broken, mostly around the multiple templates feature.

**Comment by ClaireCJS** [comment]
🎂🎂🎂 Yayyyy! 🎂🎂🎂

Glad to hear :)

**Comment by Aldaviva** [comment]
No problem, I've done that before too.

**Comment by Aldaviva** [comment]
Okay this seems a lot less broken now. There may still be issues lurking in there but it lets me create, edit, and delete multiple templates, as well as keep or discard files when Winamp is not playing.

[WinampNowPlayingToFile-3.0.0.0-b02.zip](https://github.com/user-attachments/files/19360085/WinampNowPlayingToFile-3.0.0.0-b02.zip)
